consensus/src/selection/block_generator.rs

```diff
@@ -102,7 +102,9 @@ impl<T: Operations> Generator<T> {
             },
         )?;
 
-        let tx_hashes: Vec<_> = result.txs.iter().map(|t| t.hash()).collect();
+        let tx_hashes: Vec<_> =
+            result.txs.iter().map(|t| t.inner.hash()).collect();
+        let txs: Vec<_> = result.txs.into_iter().map(|t| t.inner).collect();
```

Comment:
I'm not sure you need to specify the type you're collecting to, since it's defined in `Block::new`.